### PR TITLE
Add UI tests for embed classes

### PR DESCRIPTION
**Pull request by ayonghosh**

### CODE REVIEW

File: src/embed/search.spec.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    });
+});
+
+describe('search embed tests', () => {
+    beforeEach(() => {
+        document.body.innerHTML = getDocumentBody();
+    });
+
+    test('should render', () => {
+        const searchEmbed = new SearchEmbed(getRootEl(), defaultViewConfig);
```
Comment (jbhanu-thoughtspot) [inline src/embed/search.spec.ts:32]: Add some tests with empty defaultViewConfig, I noticed an error previously if frameParams are not added.

File: src/embed/search.spec.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+jest.spyOn(utils, 'id').mockImplementation(() => mockEmbedId);
+
+beforeAll(() => {
+    init({
+        thoughtSpotHost,
+        authType: AuthType.None,
+    });
+});
+
+describe('search embed tests', () => {
```
Comment (jbhanu-thoughtspot) [inline src/embed/search.spec.ts:26]: Add some negative tests on if params are not passed what error they should be throwing.

File: src/embed/app.spec.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    });
+});
+
+describe('pinboard/viz embed tests', () => {
+    beforeEach(() => {
+        document.body.innerHTML = getDocumentBody();
+    });
+
+    test('should render', () => {
+        const pinboardEmbed = new AppEmbed(getRootEl(), defaultViewConfig);
```
[the remark this answers is not in the document]
Comment (ayonghosh) [reply src/embed/app.spec.ts]: Done

File: src/embed/app.spec.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+const thoughtSpotHost = 'tshost';
+
+beforeAll(() => {
+    init({
+        thoughtSpotHost,
+        authType: AuthType.None,
+    });
+});
+
+describe('pinboard/viz embed tests', () => {
```
[the remark this answers is not in the document]
Comment (ayonghosh) [reply src/embed/app.spec.ts]: Done

File: src/embed/app.spec.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import { getDocumentBody, getIFrameSrc, getRootEl } from '../test/test-utils';
+
+const defaultViewConfig = {
+    frameParams: {
+        width: 1280,
+        height: 720,
+    },
+};
+const pinboardId = 'eca215d4-0d2c-4a55-90e3-d81ef6848ae0';
+const vizId = '6e73f724-660e-11eb-ae93-0242ac130002';
```
[the remark this answers is not in the document]
Comment (ayonghosh) [reply src/embed/app.spec.ts]: Done

File: src/embed/pinboard.spec.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    });
+
+    test('should render viz', () => {
+        const pinboardEmbed = new PinboardEmbed(getRootEl(), defaultViewConfig);
+        pinboardEmbed.render({
+            pinboardId,
+            vizId,
+        });
+        expect(getIFrameSrc()).toBe(
+            `http://tshost/?embedApp=true#/embed/viz/${pinboardId}/${vizId}`,
```
[the remark this answers is not in the document]
Comment (ayonghosh) [reply src/embed/pinboard.spec.ts:45]: Done

File: src/embed/app.spec.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    beforeEach(() => {
+        document.body.innerHTML = getDocumentBody();
+    });
+
+    test('should render', () => {
+        const pinboardEmbed = new AppEmbed(getRootEl(), defaultViewConfig);
+        pinboardEmbed.render({
+            pageId: Page.Search,
+        });
+        expect(getIFrameSrc()).toBe('http://tshost/?embedApp=true#/answer');
```
[the remark this answers is not in the document]
Comment (ayonghosh) [reply src/embed/app.spec.ts:31]: Done

File: src/embed/search.spec.ts (hunk 1 above)
Comment (ayonghosh) [reply src/embed/search.spec.ts:32]: Done.

File: src/embed/search.spec.ts (hunk 2 above)
Comment (ayonghosh) [reply src/embed/search.spec.ts]: Handled empty params, and added it to the basic render test. So not adding anything extra here.

**Review by jbhanu-thoughtspot — APPROVED** [review]
